src/ui/SessionExpiryModal/SessionExpiryModal.tsx

```diff
@@ -0,0 +1,77 @@
+import Cookies from 'js-cookie'
+import React, { useEffect, useState } from 'react'
+
+import Button from 'ui/Button'
+import Modal from 'ui/Modal'
+
+const ONE_MINUTE_MILLIS = 60 * 1000
+const TWO_MINUTES_MILLIS = 2 * ONE_MINUTE_MILLIS
+const THIRTY_MINUTES_MILLIS = 30 * ONE_MINUTE_MILLIS
+
+const SessionExpiryModal: React.FC = () => {
+  const [showModal, setShowModal] = useState(false)
+  const sessionExpiryTimeString = Cookies.get('session_expiry')
+
+  const checkSession = (sessionExpiryTime: Date) => {
+    const currentTime = new Date()
+    const timeLeft = sessionExpiryTime.getTime() - currentTime.getTime()
+    if (timeLeft <= TWO_MINUTES_MILLIS) {
+      setShowModal(true)
+    }
```

Comment:
Can you add in the conditional check `setShowModal`'s current state, this will block any potential infinite re-renders.